### PR TITLE
refactor: Lint project and remove deprecated code

### LINKED ISSUE #44: Project isn't linted
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server-fs-adapter/security/policy).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server-fs-adapter/issues?q=is%3Aissue).

### Current Limitation

The project isn't linted

### Feature / Enhancement Description

Lint the project to make code conform to a standard

### Example Use Case

Similar to parse-server project

### Alternatives / Workarounds

Use as is.

### 3rd Party References

N/A

### PR DESCRIPTION
**Pull request by cbaker6**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

The project isn't linted 

Closes: #44 

## Approach

- [x] Lint the project
- [x] ~~Add babel dev dependencies~~
- [x] ~~Update CI to test for successful lint~~
- [x] Replace deprecated `substr` with `substring`
- [x] Refactor `rotateEncryptionKey()` to use async/await instead of promise  
- [x] Let dependabot update action versions weekly

## Tasks

- [x] Ensure tests pass
- [ ] Change repo settings to require new tests

### CODE REVIEW

**Comment by cbaker6** [comment]
Can either revert semantic-release as suggested in #45 or merge this PR which currently uses the previous version of semantic-release to pass the CI

**Comment by mtrezza** [comment]
Could you please resolve any conflicts; and keep the semantic release version of the main branch.

**Comment by cbaker6** [comment]
My reason for opening for #45 is because it results in the following when trying to install on a local machine:

```bash
npm WARN old lockfile 
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile 
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile 
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @parse/fs-files-adapter@2.0.0
npm WARN Found: semantic-release@17.4.6
npm WARN node_modules/semantic-release
npm WARN   peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/commit-analyzer@8.0.1
npm WARN   6 more (@semantic-release/changelog, @semantic-release/npm, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/commit-analyzer@8.0.1
npm WARN node_modules/@semantic-release/commit-analyzer
npm WARN   @semantic-release/commit-analyzer@"^8.0.0" from semantic-release@17.4.6
npm WARN   node_modules/semantic-release
npm WARN   1 more (the root project)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @parse/fs-files-adapter@2.0.0
npm WARN Found: semantic-release@17.4.6
npm WARN node_modules/semantic-release
npm WARN   peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/commit-analyzer@8.0.1
npm WARN   6 more (@semantic-release/changelog, @semantic-release/npm, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer semantic-release@">=15.8.0 <18.0.0" from @semantic-release/changelog@5.0.1
npm WARN node_modules/@semantic-release/changelog
npm WARN   dev @semantic-release/changelog@"5.0.1" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @parse/fs-files-adapter@2.0.0
npm WARN Found: semantic-release@17.4.6
npm WARN node_modules/semantic-release
npm WARN   peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/commit-analyzer@8.0.1
npm WARN   6 more (@semantic-release/changelog, @semantic-release/npm, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/npm@7.1.3
npm WARN node_modules/@semantic-release/npm
npm WARN   @semantic-release/npm@"^7.0.0" from semantic-release@17.4.6
npm WARN   1 more (the root project)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @parse/fs-files-adapter@2.0.0
npm WARN Found: semantic-release@17.4.6
npm WARN node_modules/semantic-release
npm WARN   peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/commit-analyzer@8.0.1
npm WARN   6 more (@semantic-release/changelog, @semantic-release/npm, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/git@9.0.0
npm WARN node_modules/@semantic-release/git
npm WARN   dev @semantic-release/git@"9.0.0" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @parse/fs-files-adapter@2.0.0
npm WARN Found: semantic-release@17.4.6
npm WARN node_modules/semantic-release
npm WARN   peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/commit-analyzer@8.0.1
npm WARN   6 more (@semantic-release/changelog, @semantic-release/npm, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/github@7.2.3
npm WARN node_modules/@semantic-release/github
npm WARN   @semantic-release/github@"^7.0.0" from semantic-release@17.4.6
npm WARN   1 more (the root project)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @parse/fs-files-adapter@2.0.0
npm WARN Found: semantic-release@17.4.6
npm WARN node_modules/semantic-release
npm WARN   peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/commit-analyzer@8.0.1
npm WARN   6 more (@semantic-release/changelog, @semantic-release/npm, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer semantic-release@">=15.8.0 <18.0.0" from @semantic-release/release-notes-generator@9.0.3
npm WARN node_modules/@semantic-release/release-notes-generator
npm WARN   @semantic-release/release-notes-generator@"^9.0.0" from semantic-release@17.4.6
npm WARN   1 more (the root project)
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: @parse/fs-files-adapter@2.0.0
npm ERR! Found: semantic-release@21.0.1
npm ERR! node_modules/semantic-release
npm ERR!   dev semantic-release@"21.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/commit-analyzer@8.0.1
npm ERR! node_modules/@semantic-release/commit-analyzer
npm ERR!   dev @semantic-release/commit-analyzer@"8.0.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by mtrezza** [comment]
What Node version are you running? I tried with Node 16 and it worked fine.

**Comment by cbaker6** [comment]
I'm running `18.16.0`. I tried installing using `npm i -E semantic-release@21.0.1`

**Comment by mtrezza** [comment]
The main branch works it seems, so why are you trying to install the dependency explicitly?

**Comment by cbaker6** [comment]
It appears to me that dependabot allowed an incorrect update in #41 and somehow forced an update to the package-lock. The warnings and errors in https://github.com/parse-community/parse-server-fs-adapter/pull/43#issuecomment-1596273570 look explicit and state problems like `peer semantic-release@">=16.0.0 <18.0.0" from @semantic-release/github@7.2.3`.

#41 updated from `semantic-release@17.4.6` to `semantic-release@21.0.1` which are outside the peer dependency windows which is what I mentioned in #45

**Comment by cbaker6** [comment]
> The main branch works it seems, so why are you trying to install the dependency explicitly?

This PR requires babel/lint to be installed, installing using `npm i` will either revert semantic-release version to a compatible one (the current commit of this PR) or result in the errors mentioned in https://github.com/parse-community/parse-server-fs-adapter/pull/43#issuecomment-1596273570. I suspect the reasons for the this is what I mentioned in https://github.com/parse-community/parse-server-fs-adapter/pull/43#issuecomment-1596277139

**Comment by cbaker6** [comment]
The other semantic-release packages have been updated to match the peer dependency on the main branch.

I've also updated the lock file version from v1->v2 to match the server and get closer to the newer versions of node.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,23 +6,42 @@ on:
...
+          node-version: ${{ matrix.node }}
+          cache: 'npm'
+      - name: Install dependencies
+        run: npm ci
+      - run: npm run lint
+  test:
     strategy:
       matrix:
-        node: [ '14', '16', '18' ]
+        node: [ '14', '16', '18', '19' ]
```
Comment (mtrezza) [inline .github/workflows/ci.yml:30]: We'd add a Node 19 a test if we officially add Node 19 support for the adapter. This is a feature change that would need to be done in a separate PR.

**Review by mtrezza — CHANGES_REQUESTED** [review]
Just some minor stuff...

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cbaker6) [reply .github/workflows/ci.yml]: Removed 19

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -6,23 +6,42 @@ on:
   pull_request:
     branches:
       - '**'
+concurrency:
```
[the remark this answers is not in the document]
Comment (cbaker6) [reply .github/workflows/ci.yml]: Done

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (mtrezza) [inline .github/dependabot.yml:6]:
```suggestion
version: 2
updates:
  - package-ecosystem: "github-actions"
    directory: "/"
    schedule:
      interval: "weekly"
    commit-message:
      prefix: "refactor"
  - package-ecosystem: "npm"
    directory: "/"
    schedule:
      interval: "daily"
    commit-message:
      prefix: "refactor"
```

**Review by mtrezza — COMMENTED** [review]
Is it really necessary / beneficial to add babel to this repo? This won't be running in a browser but in specific Node.js environments for which we ensure compatibility via CI. Our strategy currently is to use the Node.js syntax of the lowest supported Node version for forward compatibility and it seems to work pretty well.

File: package.json [hunk 4]
```diff
@@ -19,13 +21,18 @@
...
+  ],
   "devDependencies": {
-    "@semantic-release/changelog": "5.0.1",
-    "@semantic-release/commit-analyzer": "8.0.1",
-    "@semantic-release/git": "9.0.0",
-    "@semantic-release/github": "7.2.3",
-    "@semantic-release/npm": "7.1.3",
-    "@semantic-release/release-notes-generator": "9.0.3",
+    "@babel/eslint-parser": "7.21.8",
+    "@semantic-release/changelog": "6.0.3",
```
[the remark this answers is not in the document]
Comment (cbaker6) [reply package.json:30]: I'm not sure I understand, is this a note for you, or me to try something? If it's for me, what are the steps I need to do to validate this?

[the remark this answers is not in the document]
Comment (mtrezza) [reply package.json:30]: Same as https://github.com/parse-community/parse-server/pull/8643#issuecomment-1597294393

[the remark this answers is not in the document]
Comment (cbaker6) [reply package.json:30]: From what I can tell, dry-runs seem to be for people who access to the original repo, I don't see much about parameters to point to my fork. Below is what I get:

```bash
[3:55:46 PM] [semantic-release] › ℹ  Running semantic-release version 21.0.1
Running on branch: main
Changelog file output to: ./CHANGELOG.md
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/changelog"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/git"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/changelog"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/git"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/github"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[3:55:47 PM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[3:55:49 PM] [semantic-release] › ⚠  Run automated release from branch main on repository https://github.com/parse-community/parse-server-fs-adapter in dry-run mode
[3:55:50 PM] [semantic-release] › ✘  The command "git push --dry-run --no-verify https://github.com/parse-community/parse-server-fs-adapter HEAD:main" failed with the error message remote: Permission to parse-community/parse-server-fs-adapter.git denied to cbaker6.
fatal: unable to access 'https://github.com/parse-community/parse-server-fs-adapter/': The requested URL returned error: 403.
[3:55:50 PM] [semantic-release] › ⚠  Skip step "fail" of plugin "@semantic-release/github" in dry-run mode
[3:55:50 PM] [semantic-release] › ✘  EGITNOPERMISSION Cannot push to the Git repository.
semantic-release cannot push the version tag to the branch main on the remote Git repository with URL https://github.com/parse-community/parse-server-fs-adapter.

This can be caused by:

    * a misconfiguration of the repositoryUrl (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#repositoryurl) option
    * the repository being unavailable
    * or missing push permission for the user configured via the Git credentials on your CI environment (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/ci-configuration.md#authentication)

SemanticReleaseError: Cannot push to the Git repository.
    at default (file:///Users/MacJr/Documents/github/parse-server-fs-adapter/node_modules/semantic-release/lib/get-error.js:6:10)
    at run (file:///Users/MacJr/Documents/github/parse-server-fs-adapter/node_modules/semantic-release/index.js:101:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Module.default (file:///Users/MacJr/Documents/github/parse-server-fs-adapter/node_modules/semantic-release/index.js:275:22)
    at async default (file:///Users/MacJr/Documents/github/parse-server-fs-adapter/node_modules/semantic-release/cli.js:55:5) {
  code: 'EGITNOPERMISSION',
  details: '**semantic-release** cannot push the version tag to the branch `main` on the remote Git repository with URL `https://github.com/parse-community/parse-server-fs-adapter`.\n' +
    '\n' +
    'This can be caused by:\n' +
    ' - a misconfiguration of the [repositoryUrl](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#repositoryurl) option\n' +
    ' - the repository being unavailable\n' +
    ' - or missing push permission for the user configured via the [Git credentials on your CI environment](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/ci-configuration.md#authentication)',
  semanticRelease: true
}%                                                    
```

[the remark this answers is not in the document]
Comment (cbaker6) [reply package.json:30]: The log I posted shows some `dry-run` comments:

```bash
[3:55:49 PM] [semantic-release] › ⚠  Run automated release from branch main on repository https://github.com/parse-community/parse-server-fs-adapter in dry-run mode
[3:55:50 PM] [semantic-release] › ✘  The command "git push --dry-run --no-verify https://github.com/parse-community/parse-server-fs-adapter HEAD:main" failed with the error message remote: Permission to parse-community/parse-server-fs-adapter.git denied to cbaker6.
fatal: unable to access 'https://github.com/parse-community/parse-server-fs-adapter/': The requested URL returned error: 403.
[3:55:50 PM] [semantic-release] › ⚠  Skip step "fail" of plugin "@semantic-release/github" in dry-run mode
```

I also changed dry-run in `release.config.js`, but it did the same thing.

[the remark this answers is not in the document]
Comment (mtrezza) [reply package.json:30]: I see, it tries a git push dry-run, hence the error. You could try to disable that if possible or just remove the git step temporarily from the config file, so it doesn't try to push a tag.

[the remark this answers is not in the document]
Comment (cbaker6) [reply package.json:30]: I can't get this to work... It seems since you have access, it will be easier for you to attempt the dry run since you are familiar with the semantic-release package and I am not.

[the remark this answers is not in the document]
Comment (mtrezza) [reply package.json:30]: I'll take a look

**Comment by cbaker6** [comment]
Ready for review...

In the repo settings you should add protections for the following passing tests:
- ci/Node 14
- ci/Node 16
- ci/Node 18
- ci/Node 19

Remove the following as they are replaced by the aforementioned tests:
- test (14)
- test (16)
- test (18)

**Review by mtrezza — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.0.0](https://github.com/parse-community/parse-server-fs-adapter/releases/tag/3.0.0)
